src/js/experimental/media-rendition-listbox.js

```diff
@@ -0,0 +1,153 @@
+import MediaChromeListbox from './media-chrome-listbox.js';
+import './media-chrome-option.js';
+import { window, document } from '../utils/server-safe-globals.js';
+import { parseRenditionList } from '../utils/utils.js';
+import { MediaUIAttributes, MediaUIEvents } from '../constants.js';
+
+const slotTemplate = document.createElement('template');
+slotTemplate.innerHTML = /*html*/`
+  <style>
+    media-chrome-option {
+      white-space: var(--media-rendition-listbox-white-space, nowrap);
+    }
+  </style>
+`;
+
+/**
+ * @attr {string} mediarenditionenabled - (read-only) Set to the enabled rendition.
+ * @attr {string} mediarenditionlist - (read-only) Set to the rendition list.
+ *
+ * @cssproperty --media-rendition-listbox-white-space - `white-space` of playback rate list item.
+ */
+class MediaRenditionListbox extends MediaChromeListbox {
+  #autoOption;
+  #renditionList = [];
+
+  static get observedAttributes() {
+    return [
+      ...super.observedAttributes,
+      MediaUIAttributes.MEDIA_RENDITION_LIST,
+      MediaUIAttributes.MEDIA_RENDITION_ENABLED,
+    ];
+  }
+
+  constructor() {
+    super({ slotTemplate });
+
+    const autoOption = document.createElement('media-chrome-option');
+
+    autoOption.part.add('option');
+    autoOption.value = '';
+    autoOption.textContent = 'Auto';
+    this.#autoOption = autoOption;
+  }
+
+  attributeChangedCallback(attrName, oldValue, newValue) {
+
+    if (attrName === MediaUIAttributes.MEDIA_RENDITION_ENABLED && oldValue !== newValue) {
+      this.value = newValue;
+
+    } else if (attrName === MediaUIAttributes.MEDIA_RENDITION_LIST && oldValue !== newValue) {
+
+      this.#renditionList = parseRenditionList(newValue);
+      this.#render();
+    }
+
+    super.attributeChangedCallback(attrName, oldValue, newValue);
+  }
+
+  connectedCallback() {
+    this.addEventListener('change', this.#onChange);
+
+    super.connectedCallback();
+  }
+
+  disconnectedCallback() {
+    this.removeEventListener('change', this.#onChange);
+
+    super.disconnectedCallback();
+  }
+
+  get mediaRenditionList() {
+    return this.#renditionList;
+  }
+
+  set mediaRenditionList(list) {
+    this.removeAttribute(MediaUIAttributes.MEDIA_RENDITION_LIST);
```

Comment:
Why remove the attr here?

Reply:
this is related to @cjpillsbury feedback of making some rendition related API only via props (not reflecting to attrs) but still accept attributes.

the prop / attribute is not reflected anymore so we used this behavior for mediaController.fullscreenElement in removing the attribute if the property was used.

the alternative is if there was ever an attribute set to also keep updating the attribute if ever the property was set but would make it more complex I think.

Reply:
Yeah, that makes sense. Complex values shouldn't be sprouted because you're not going to style things with them, and even simplifying the complex values would open up other complexity.

Deleting it is still form of sprouting, but I guess things like `autoplay` already do that.

Reply:
yes, I can see that. do you think it's better to leave it as is? not remove the attribute.
I'm fine with both I think, it's a pretty rare case